### PR TITLE
Change default parquet format version to v2_latest, add a setting to controle the version

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Use parquet format v2 instead of v1 in output format by default. Add setting `output_format_parquet_version` to control parquet version, possible values `v1_0`, `v2_4`, `v2_6`, `v2_latest` (default)

### CODE REVIEW

**Comment by Avogar** [comment]
Let's see if it will be ok to switch to Parquet v2

File: docs/en/operations/settings/settings-formats.md [hunk 1]
```diff
@@ -1102,6 +1102,12 @@ Use Parquet FIXED_LENGTH_BYTE_ARRAY type instead of Binary/String for FixedStrin
 
 Enabled by default.
 
+### output_format_parquet_version {#output_format_parquet_version}
+
+The version of Parquet format used in output format. Supported versions: `v1_0`, `v2_4`, `v2_6` and `v2_latest`.
```
Comment (alexey-milovidov) [inline docs/en/operations/settings/settings-formats.md:1107]: Is `v` and underscore natural or expected?

Why don't we use `2.4`, `2.latest`?

Comment (Avogar) [reply docs/en/operations/settings/settings-formats.md:1107]: I just used same names as it's in arrow parquet library. But let's change to version with dots.

File: src/Core/SettingsChangesHistory.h [hunk 2]
```diff
@@ -80,7 +80,8 @@ namespace SettingsChangesHistory
 /// It's used to implement `compatibility` setting (see https://github.com/ClickHouse/ClickHouse/issues/35972)
 static std::map<ClickHouseVersion, SettingsChangesHistory::SettingsChanges> settings_changes_history =
 {
-    {"23.3", {{"input_format_json_ignore_unknown_keys_in_named_tuple", false, true, "Improve parsing JSON objects as named tuples"}}},
+    {"23.3", {{"output_format_parquet_version", "v1_0", "v2_latest", "Use latest Parquet format version for output format"},
```
Comment (Avogar) [inline src/Core/SettingsChangesHistory.h:83]:
```suggestion
    {"23.3", {{"output_format_parquet_version", "1.0", "2.latest", "Use latest Parquet format version for output format"},
```

File: src/Core/Settings.h [hunk 3]
```diff
@@ -856,6 +856,7 @@ class IColumn;
     M(UInt64, output_format_parquet_row_group_size, 1000000, "Row group size in rows.", 0) \
     M(Bool, output_format_parquet_string_as_string, false, "Use Parquet String type instead of Binary for String columns.", 0) \
     M(Bool, output_format_parquet_fixed_string_as_fixed_byte_array, true, "Use Parquet FIXED_LENGTH_BYTE_ARRAY type instead of Binary for FixedString columns.", 0) \
+    M(ParquetVersion, output_format_parquet_version, "v2_latest", "Parquet format version for output format. Supported versions: v1_0, v2_4, v2_6 and v2_latest (default)", 0) \
```
Comment (Avogar) [inline src/Core/Settings.h:859]:
```suggestion
    M(ParquetVersion, output_format_parquet_version, "2.latest", "Parquet format version for output format. Supported versions: 1.0, 2.4, 2.6 and 2.latest (default)", 0) \
```

File: docs/en/operations/settings/settings-formats.md [hunk 4]
```diff
@@ -1102,6 +1102,12 @@ Use Parquet FIXED_LENGTH_BYTE_ARRAY type instead of Binary/String for FixedStrin
...
+
+Default value: `v2_latest`.
```
Comment (Avogar) [inline docs/en/operations/settings/settings-formats.md:1109]:
```suggestion
Default value: `2.latest`.
```

File: docs/en/operations/settings/settings-formats.md (hunk 1 above)
Comment (Avogar) [reply docs/en/operations/settings/settings-formats.md:1107]:
```suggestion
The version of Parquet format used in output format. Supported versions: `1.0`, `2.4`, `2.6` and `2.latest`.
```

File: docs/en/interfaces/formats.md [hunk 5]
```diff
@@ -1971,7 +1971,8 @@ To exchange data with Hadoop, you can use [HDFS table engine](/docs/en/engines/t
...
 - [input_format_parquet_skip_columns_with_unsupported_types_in_schema_inference](/docs/en/operations/settings/settings-formats.md/#input_format_parquet_skip_columns_with_unsupported_types_in_schema_inference) - allow skipping columns with unsupported types while schema inference for Parquet format. Default value - `false`.
-- [output_format_parquet_fixed_string_as_fixed_byte_array](/docs/en/operations/settings/settings-formats.md/#output_format_parquet_fixed_string_as_fixed_byte_array) -  use Parquet FIXED_LENGTH_BYTE_ARRAY type instead of Binary/String for FixedString columns. Default value - `true`.
+- [output_format_parquet_fixed_string_as_fixed_byte_array](/docs/en/operations/settings/settings-formats.md/#output_format_parquet_fixed_string_as_fixed_byte_array) - use Parquet FIXED_LENGTH_BYTE_ARRAY type instead of Binary/String for FixedString columns. Default value - `true`.
+- [output_format_parquet_version](/docs/en/operations/settings/settings-formats.md/#output_format_parquet_version) - The version of Parquet format used in output format. Default value - `v2_latest`.
```
Comment (Avogar) [inline docs/en/interfaces/formats.md:1975]:
```suggestion
- [output_format_parquet_version](/docs/en/operations/settings/settings-formats.md/#output_format_parquet_version) - The version of Parquet format used in output format. Default value - `2.latest`.
```

**Review by evillique — APPROVED** [review]

**Pull request merged**